hack/gen-push-swagger-markdown.sh

```diff
@@ -0,0 +1,180 @@
+    echo "      will be used, git ssh connection otherwise."
+    echo ""
+    echo "If --wait specified:"
+    echo "    * wait for docs-csm RPM package to be published at https://artifactory.algol60.net"
+    echo "    * ARTIFACTORY_USERNAME and ARTIFACTORY_TOKEN environment variables must be set"
+    exit 1
+}
+
+function error() {
+    echo "${1}"
+    exit 1
+}
+
+function wait_for_docs_csm_publish() {
+    tag="${1}"
+    filename="docs-csm-${tag#v}-1.noarch.rpm"
+    # 120 attempts with 10 sec wait gives ~ 20 minutes for package to be published.
+    attempts=120
+    wait=10
+    echo "Waiting for ${filename} to be published ..."
+    counter=0
+    while [ $counter -le $attempts ]; do
+        num_results=$(curl -Ss -u "${ARTIFACTORY_USERNAME}:${ARTIFACTORY_TOKEN}" "https://artifactory.algol60.net/artifactory/api/search/artifact?name=${filename}&repos=csm-rpms" | jq -r '.results | length')
+        if [ $((num_results)) -eq 0 ]; then
+            echo "Attempt ${counter} / ${attempts}: not published yet"
+        else
+            echo "Package ${filename} was successfully published."
+            return 0
+        fi
+        counter=$(($counter + 1))
+        sleep $wait
+    done
+    echo "ERROR: giving up after ${attempts} attempts."
+    exit 2
+}
+
+function increment_push_tag() {
+    wait="${1}"
+    last_tag="$(git describe --tags | awk -F- '{print $1}')"
+    echo "${last_tag}" | grep -E -q -x 'v[0-9]+\.[0-9]+\.[0-9]+' || error "ERROR: Unable to compute version increment. Latest tag ${last_tag} of docs-csm repo is not in vX.Y.Z format."
+    new_tag=$(echo "${last_tag}" | awk -F. '{print $1 "." $2 "." (int($3)+1) }')
+    echo "Tagging docs-csm with new tag ${new_tag} ..."
+    git tag "${new_tag}"
+    echo "Pushing tag ${new_tag} of docs-csm ..."
+    git push -q origin "${new_tag}"
+    if [ "${wait}" == "1" ]; then
+        wait_for_docs_csm_publish "${new_tag}"
+    fi
+}
+
+docs_csm_branch="${1}"; shift || usage
+push=0
+wait=0
+
+while [ -n "${1}" ]; do
+    case "${1}" in
+        --push)
+            push=1
+            ;;
+        --wait)
+            wait=1
+            ;;
+        *)
+            usage
+            ;;
+    esac
+    shift
+done
+
+yq="docker run --rm -i artifactory.algol60.net/docker.io/mikefarah/yq:4"
+manifest_dir=$(realpath "${0}")
+manifest_dir=$(dirname "${manifest_dir}")
+manifest_dir=$(realpath "${manifest_dir}/../")/manifests
+dest_dir="$(mktemp -d)"
+trap 'echo "Cleaning up ..."; docker rm -f widdershins >/dev/null; rm -rf "${dest_dir}"' EXIT
+
+mkdir -p "${dest_dir}"
+echo "Cloning ${docs_csm_branch} branch of docs-csm ..."
+if [ -z "${GITHUB_APP_INST_TOKEN}" ]; then
+    git clone -q --branch "${docs_csm_branch}" git@github.com:Cray-HPE/docs-csm.git "${dest_dir}/docs-csm"
+else
+    bash_settings="$-"
+    # Avoid token exposure in tracing log
+    set +x
+    # If tag protection is enabled on the repo, tag can not be pushed using deploy key (via ssh protocol). Application with
```

Comment:
@phalseth-hpe It appears our `docs-csm` repo has tag protection feature enabled. I.e., write permission is insufficient to push a tag - it must be Admin or Maintain permission. This means regular deploy key can not be used, because it can only provide write permission. Instead, we need to use Github application, which can only authenticate to git using app installation tokens, via https protocol.